sdk/servicebus/azure-servicebus/azure/servicebus/aio/_base_handler_async.py

```diff
@@ -114,23 +115,42 @@ async def __aenter__(self):
     async def __aexit__(self, *args):
         await self.close()
 
-    async def _handle_exception(self, exception, **kwargs):
-        error, error_need_close_handler, error_need_raise = \
-            _create_servicebus_exception(_LOGGER, exception, self, **kwargs)
-        if error_need_close_handler:
+    async def _handle_exception(self, exception):
+        # pylint: disable=protected-access
+        error = _create_servicebus_exception(_LOGGER, exception)
+
+        try:
+            # If SessionLockLostError or ServiceBusConnectionError happen when a session receiver is running,
+            # the receiver can no longer be used and should create a new session receiver
```

Comment:
```suggestion
            # the receiver should no longer be used and should create a new session receiver
```
So the one thing I'd note here (which the above tweak gets at but doesn't complete because <waves fist at github> no multi-line suggestions yet; is our motivation for disallowing this: namely that there are pitfalls WRT both next session IDs, and the diversity of session failure modes, that motivates us to disallow this. (as opposed to the original verbage of "can no longer be used" which to me implied that there's a service reason for it.)

Reply:
put your words into the comments++